src/mapping/circular_to_cartesian.hpp

```diff
@@ -163,73 +162,33 @@ class CircularToCartesian
     }
 
     /**
-     * @brief Compute the (1,1) coefficient of the Jacobian matrix.
-     *
-     * For a mapping given by @f$ \mathcal{F} : (r,\theta)\mapsto (x,y) @f$, the
-     * (1,1) coefficient of the Jacobian matrix is given by @f$ \frac{\partial x}{\partial r} @f$.
-     *
+     * @brief Compute the (i,j) coefficient of the Jacobian matrix.
+     * 
      * @param[in] coord
      *              The coordinate where we evaluate the Jacobian matrix.
      *
-     * @return A double with the value of the (1,1) coefficient of the Jacobian matrix.
-     */
-    KOKKOS_FUNCTION double jacobian_11(Coord<R, Theta> const& coord) const
-    {
-        const double theta = ddc::get<Theta>(coord);
-        return Kokkos::cos(theta);
-    }
-
-    /**
-     * @brief Compute the (1,2) coefficient of the Jacobian matrix.
-     *
-     * For a mapping given by @f$ \mathcal{F} : (r,\theta)\mapsto (x,y) @f$, the
-     * (1,2) coefficient of the Jacobian matrix is given by @f$ \frac{\partial x}{\partial \theta} @f$.
-     *
-     * @param[in] coord
-     *              The coordinate where we evaluate the Jacobian matrix.
-     *
-     * @return A double with the value of the (1,2) coefficient of the Jacobian matrix.
-     */
-    KOKKOS_FUNCTION double jacobian_12(Coord<R, Theta> const& coord) const
-    {
-        const double r = ddc::get<R>(coord);
-        const double theta = ddc::get<Theta>(coord);
-        return -r * Kokkos::sin(theta);
-    }
-
-    /**
-     * @brief Compute the (2,1) coefficient of the Jacobian matrix.
-     *
-     * For a mapping given by @f$ \mathcal{F} : (r,\theta)\mapsto (x,y) @f$, the
-     * (2,1) coefficient of the Jacobian matrix is given by @f$ \frac{\partial y}{\partial r} @f$.
-     *
-     * @param[in] coord
-     *              The coordinate where we evaluate the Jacobian matrix. .
-     *
-     * @return A double with the value of the (2,1) coefficient of the Jacobian matrix.
+     * @return A double with the value of the (i,j) coefficient of the Jacobian matrix.
      */
-    KOKKOS_FUNCTION double jacobian_21(Coord<R, Theta> const& coord) const
+    template <class IndexTag1, class IndexTag2>
+    KOKKOS_FUNCTION double jacobian_component(Coord<R, Theta> const& coord) const
     {
```

Comment:
To check the type you should use one of the following:
```suggestion
    {
        static_assert(ddc::in_tags_v<IndexTag1, VectorIndexSet<X, Y>>);
        static_assert(ddc::in_tags_v<IndexTag2, VectorIndexSet<R_cov, Theta_cov>>);
```

```suggestion
    {
        static_assert(ddc::in_tags_v<IndexTag1, ddc::to_type_seq_t<CoordResult>>>);
        static_assert(ddc::in_tags_v<IndexTag2, vector_index_set_dual_t<ddc::to_type_seq_t<CoordArg>>>);
```

```suggestion
    {
        static_assert(std::is_same_v<IndexTag1, X> || std::is_same_v<IndexTag1, Y>);
        static_assert(std::is_same_v<IndexTag2, R_cov> || std::is_same_v<IndexTag2, Theta_cov>);
```

I am not sure which is the most readable. Probably the first one.
The meaning can be clarified with a string. E.g:
```suggestion
    {
        static_assert(ddc::in_tags_v<IndexTag1, VectorIndexSet<X, Y>>, "The first dimension tag should represent a Cartesian coordinate");
```